presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -79,9 +81,10 @@ public Analysis analyze(Statement statement, boolean isDescribe)
         Analysis analysis = new Analysis(rewrittenStatement, parameters, isDescribe);
         StatementAnalyzer analyzer = new StatementAnalyzer(analysis, metadata, sqlParser, accessControl, session, warningCollector);
         analyzer.analyze(rewrittenStatement, Optional.empty());
+        analyzeForUtilizedColumns(analysis, analysis.getStatement());
 
         // check column access permissions for each table
-        analysis.getTableColumnReferences().forEach((accessControlInfo, tableColumnReferences) ->
+        analysis.getTableColumnReferencesForAccessControl(isCheckAccessControlOnUtilizedColumnsOnly(session)).forEach((accessControlInfo, tableColumnReferences) ->
```

Comment:
A better style would be to send the session object to the function and do the flag check there for better encapsulation.

Reply:
Done